src/notebook/output-area/widget.ts

```diff
@@ -466,9 +469,12 @@ namespace OutputAreaWidget {
 
       // Remove any existing content.
       let layout = widget.layout as PanelLayout;
-      for (let i = 0; i < layout.childCount(); i++) {
-        layout.childAt(0).dispose();
+      let child = layout.childAt(1);
```

Comment:
This seems brittle to me. Any change to child order/count will require these indices to at least be looked at. What about a simple widget subclass which exposes the widgets you care about as getters, and manages the layout internally?